### PR TITLE
Creating models and performing CRUD

### PR DESCRIPTION
**Pull request by d2ndjim**
- Created model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
- Set up associations between models.
- Created at least one user.
- Create at least 4 posts written by one of the users created.
- Create at least 6 posts comments for one of the posts created.
- Added a method that returns the 3 most recent posts for a given user to User model
- Added a method that updates the posts counter for a user and a method which returns the 5 most recent comments for a given post to the Post model
- Added a method that updates the comments counter for a post in the Comment model
- Added a method that updates the likes counter for a post in the likes model
- Used rails console to test methods and perform CRUD operations

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts
```
Comment (arslanbisharat) [inline app/models/user.rb]: - [ ] Please make sure the post has `foreign_key: 'author_id'`

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  has_many :comments
```
Comment (arslanbisharat) [inline app/models/user.rb]: - [ ] Please make sure the comments has `foreign_key: 'author_id'`

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  has_many :likes
```
Comment (arslanbisharat) [inline app/models/user.rb]: - [ ] Please make sure the likes has `foreign_key: 'author_id'`

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
Hi @d2ndjim, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The Pull Request has a proper title and description. ✔️
- You have added a descriptive Readme file. Good Job. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please make sure you right descriptive comments. Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. :x:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KDlamini — APPROVED** [review]
### Hi @d2ndjim 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ Models setup correctly
✔️ Good PR title
✔️ Good PR description
✔️ Correctly use of GitFlow
✔️ No linter errors

# Status: Approved ✔️ 

simon

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**